### PR TITLE
[GDAL] Update to 2.3.0

### PR DESCRIPTION
**Pull request by mloskot**
Update portfile.cmake for easier version/checksum maintenance.
Update and rename 0001 patch fixing debug build.
Update no-my-bool.patch following source file changes.
Remove no patches which are no longer required due to corresponding
changes in the GDAL source code.

Supports PROJ 4.9.x and 5.x - the latter not in vcpkg ports.

### CODE REVIEW

File: ports/gdal/no-my-bool.patch [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
- 
+diff -Nuar a/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp b/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp
+--- a/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp	2018-05-04 09:09:26.000000000 +0200
++++ b/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp	2018-05-11 23:07:44.284558300 +0200
+@@ -235,7 +235,7 @@
+         // Enable automatic reconnection
+         // Must be called after mysql_real_connect() on MySQL < 5.0.19
+         // and at any point on more recent versions.
+-        my_bool reconnect = 1;
++        bool reconnect = 1;
```
Comment (rouault) [inline ports/gdal/no-my-bool.patch]: shouldn't all those mysql related patches be upstreamed ? (would probably require some conditionalizing about the version)

**Comment by rouault** [comment]
> Requires PROJ update to 5.0.1 or later.

There's perhaps an ambiguity in the way the NEWS has been written. The use of PROJ5 API requires PROJ 5.0.1 or later (that is: PROJ 5.0.0 is not supported due to bugs/missing functions). But GDAL 2.3 can still build and work against proj 4.9.X .

**Comment by mloskot** [comment]
@rouault 
>>  Requires PROJ update to 5.0.1 or later.
>
> There's perhaps an ambiguity in the way the NEWS has been written. 

I admit I have read the NEWS quite literally and didn't double check my understanding.

I will clarify the issue as having no blockers.

File: ports/gdal/no-my-bool.patch (hunk 1 above)
Comment (mloskot) [reply ports/gdal/no-my-bool.patch]: Patching upstream is always highly recommended. I'm not the author of those patches and I don't know why they have not been proposed to GDAL. However, I am going to submit,

**Comment by ras0219-msft** [comment]
Looks great to me, thanks for the PR!

**Comment by mloskot** [comment]
@ras0219-msft Thanks!

File: ports/gdal/no-my-bool.patch (hunk 1 above)
Comment (mloskot) [reply ports/gdal/no-my-bool.patch]: @rouault I just submitted the patch to GDAL https://github.com/OSGeo/gdal/pull/631

**Pull request merged**